tests/ingestIndexTestBase.py

```diff
@@ -193,7 +189,8 @@ def tearDownClass(cls):
 
     @classmethod
     def setUpClass(cls):
-        cls.outPath = tempfile.mkdtemp()
+        cls.outDir = tempfile.TemporaryDirectory()
```

Comment:
We try to avoid using this in tests because it can lead to test failures over NFS caused by TemporaryDirectory not using the ignore_errors flag in shutil when it automatically runs the cleanup (because of nfs lock files hanging around). You might get away with it.

Reply:
Oh no!  Okay, we had a mix before and I thought that `mkdtemp` was old and `TemporaryDirectory` was new.  I can move everything over to `mkdtemp`.  The problem I was trying to solve was that the temporary directories aren't being cleaned up.  (These _might_ be fine because they all go in `/tmp` and who mounts `/tmp` over nfs?)

Reply:
Other people may be excited about fancy case syntax, but [`ignore_cleanup_errors`](https://docs.python.org/3/library/tempfile.html?highlight=tempfile#tempfile.TemporaryDirectory) is what I'm really looking forward to.

Reply:
Ooh. They've finally fixed it to allow you to say you don't care. If only someone would write a 3.10 migration RFC.

Reply:
Hmm. Don't we have `lsst.utils.tests.temporaryDirectory` that tries to fix this?

Reply:
Question: should I move this all back to mkdtemp and add the robust cleanup now, or let this slide until we get to 3.10?  I think it's fine because `/tmp` shouldn't be nfs mounted, but that doesn't mean it can't be nfs mounted.

Reply:
:+1: to `lsst.utils.tests.temporaryDirectory` as the solution for now.

Reply:
Hmmm.  That's a context manager so does it work against setup/teardown which we need here?

Reply:
🤔 You are probably right that /tmp is not going to mess us up. I think the problem we had before was from people creating temp directories within the `tests/` directory (which is still common).

Reply:
I don't see any notes about temporary directory usage in the dev guide, and I wasn't aware of that problem. Should we add a section about that to the python testing guide?